01_explore_data/nbconverted/explore_cancer_gene_sets.py

```diff
@@ -0,0 +1,388 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Explore cancer gene sets
+# 
+# In `00_download_data/2_download_cancer_gene_data.ipynb`, we downloaded and integrated several different sets of cancer-associated genes. In this script, we want to explore the similarities and differences between these gene sets in more detail.
+# 
+# We see that the results of our classification experiments are somewhat dependent on which gene set or sets we decide to use, so understanding the similarities and differences between the gene sets will help us to put these in context.
+
+# In[1]:
+
+
+import sys
+from pathlib import Path
+
+import pandas as pd
+
+import mpmp.config as cfg
+import mpmp.utilities.data_utilities as du
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+
+# In[2]:
+
+
+# load datasets
+vogelstein_df = du.load_vogelstein()
+vogelstein_df.head()
+
+
+# In[3]:
+
+
+cosmic_df = du.load_cosmic()
+cosmic_df.head()
+
+
+# In[4]:
+
+
+# load Bailey et al. data from excel file
+# this is the same as the code in 00_download_data/2_download_cancer_gene_set.ipynb
+class_df = pd.read_excel(
+    cfg.bailey_raw_file,
+    engine='openpyxl', sheet_name='Table S1', index_col='KEY', header=3
+)
+class_df.drop(
+    class_df.columns[class_df.columns.str.contains('Unnamed')],
+    axis=1, inplace=True
+)
+class_df.rename(columns={'Tumor suppressor or oncogene prediction (by 20/20+)':
+                         'classification'},
+                inplace=True)
+
+bailey_df = (
+    class_df[((class_df.Cancer == 'PANCAN') &
+             (~class_df.classification.isna()))]
+).copy()
+
+bailey_df.head()
+
+
+# ### Overlap between COSMIC/Bailey/Vogelstein
+# 
+# Is COSMIC a strict subset of the Bailey and Vogelstein cancer driver datasets? Or are there genes in the latter two that are not in COSMIC?
+
+# In[5]:
+
+
+vogelstein_genes = set(vogelstein_df.gene.values)
+cosmic_genes = set(cosmic_df.gene.values)
+bailey_genes = set(bailey_df.Gene.values)
+
+
+# In[6]:
+
+
+from venn import venn
```

Comment:
Nice plot.

I assume you've already checked this, but I know that gene id mapping can always be tricky. So you made sure that the differences are not due to some mapping issue?

Reply:
By "gene id mapping", do you mean the mapping of genes to GO terms? Or the mapping of gene symbols to Entrez IDs?

We're doing both in this script - just want to check which you've found tricky (or if both are tricky). For the gene -> GO term mapping I basically just followed a tutorial, and the symbol -> Entrez mapping I've done several times before so I'm a bit more familiar.

Reply:
Sorry I was referring to the gene symbol to entrez not always being 1-1. But I'm sure you checked I was double checking just in case

Reply:
I haven't checked super closely, I'm relying here on the gene mapping that was developed for Cognoma (see, e.g. [this PR](https://github.com/cognoma/cancer-data/pull/12)) which Greg has also used in most of his classification repos. As far as I can tell they dropped ambiguous symbol -> ID mappings from their maps, and almost all of the genes here seem to map to IDs, so I think I'm okay with that since this is a quick one-off analysis.

I don't think the differences in enriched GO terms that we see here are due to gene mapping ambiguities -- I think they're almost certainly due to actual differences in the genes in each gene set -- but it's still worth thinking about!